pymc_marketing/clv/models/modified_beta_geo.py

```diff
@@ -0,0 +1,316 @@
+        t: np.ndarray | pd.Series | pt.TensorVariable,
+        frequency: np.ndarray | pd.Series | pt.TensorVariable,
+        recency: np.ndarray | pd.Series | pt.TensorVariable,
+        T: np.ndarray | pd.Series | pt.TensorVariable,
+    ) -> xarray.DataArray:
+        r"""Compute the expected number of purchases for a customer.
+
+        This is a deprecated method and will be removed in a future release.
+        Please use `BetaGeoModel.expected_purchases` instead.
+        """
+        warnings.warn(
+            "Deprecated method. Use 'expected_purchases' instead.",
+            FutureWarning,
+            stacklevel=1,
+        )
+
+        t = np.asarray(t)
+        if t.size != 1:
+            t = to_xarray(customer_id, t)
+
+        T = np.asarray(T)
+        if T.size != 1:
+            T = to_xarray(customer_id, T)
+
+        # print(customer_id)
+        x, t_x = to_xarray(customer_id, frequency, recency)
+
+        a, b, alpha, r = self._unload_params()
+
+        hyp_term = hyp2f1(r + x, b + x + 1, a + b + x, t / (alpha + T + t))
+        first_term = (a + b + x) / (a - 1)
+        second_term = 1 - hyp_term * ((alpha + T) / (alpha + t + T)) ** (r + x)
+        numerator = first_term * second_term
+
+        denominator = 1 + (a / (b + x)) * ((alpha + T) / (alpha + t_x)) ** (r + x)
+
+        return (numerator / denominator).transpose(
+            "chain", "draw", "customer_id", missing_dims="ignore"
+        )
+
+    def expected_purchases(
+        self,
+        data: pd.DataFrame | None = None,
+        *,
+        future_t: int | np.ndarray | pd.Series | None = None,
+    ) -> xarray.DataArray:
+        r"""Compute the expected number of future purchases across *future_t* time periods given *recency*, *frequency*, and *T* for each customer.
+
+        The *data* parameter is only required for out-of-sample customers.
+
+        Adapted from equation (10) in [1]_, and *lifetimes* package:
+        https://github.com/CamDavidsonPilon/lifetimes/blob/41e394923ad72b17b5da93e88cfabab43f51abe2/lifetimes/fitters/beta_geo_fitter.py#L201
+
+        Parameters
+        ----------
+        future_t : int, array_like
+            Number of time periods to predict expected purchases.
+        data : ~pandas.DataFrame
+            Optional dataframe containing the following columns:
+
+            * `customer_id`: Unique customer identifier
+            * `frequency`: Number of repeat purchases
+            * `recency`: Time between the first and the last purchase
+            * `T`: Time between first purchase and end of observation period; model assumptions require T >= recency
+
+        References
+        ----------
+        .. [1] Fader, Peter S., Bruce G.S. Hardie, and Ka Lok Lee (2005a),
+            "Counting Your Customers the Easy Way: An Alternative to the
+            Pareto/NBD Model," Marketing Science, 24 (2), 275-84.
+            https://www.brucehardie.com/papers/bgnbd_2004-04-20.pdf
+
```

Comment:
These docstrings need to be updated to reflect the current model.

Reply:
Updated